evalml/pipelines/pipeline_base.py

```diff
@@ -709,6 +709,8 @@ def repr_component(parameters):
         parameters_repr = ", ".join(
             [
                 f"'{component}':{{{repr_component(parameters)}}}"
+                if component != "DFS Transformer"
```

Comment:
Need to exclude the `DFS Transformer` since the `__repr__` of a feature is not executable.

Reply:
could this ever be problematic if there's separate `Numeric Pipeline` vs `Categorical Pipeline`? I forget if dfs transformer comes before we split pipelines, though, so I definitely might not have all the info here.

Reply:
Same question goes for `has_dfs = "DFS Transformer" in element.component_graph.compute_order` below

Reply:
the `DFS Transformer` always comes first so I don't believe it should be a problem (and it'll always be named `DFS Transformer`